hbase-server/src/test/java/org/apache/hadoop/hbase/master/http/TestHbckMetricsResource.java

```diff
@@ -0,0 +1,422 @@
+  private static final String port = "16010";
+  private static final String hostStartCode = "123456789";
+  private static final String path1 = "hdfs://path1";
+  private static final String path2 = "hdfs://path2";
+  private static final String metaRegionID = FIRST_META_REGIONINFO.getEncodedName();
+  private static final String metaTableName = FIRST_META_REGIONINFO.getTable().getNameAsString();
+
+  // Various Keys in HBCK JSON Response.
+  private static final String quoteColon = "\":";
+  private static final String quote = "\"";
+  private static final String regionId = quote + "region_id" + quoteColon;
+  private static final String regionHdfsPath = quote + "region_hdfs_path" + quoteColon;
+  private static final String rsName = quote + "rs_name" + quoteColon;
+  private static final String hostName = quote + "host_name" + quoteColon;
+  private static final String hostPort = quote + "host_port" + quoteColon;
+  private static final String startCode = quote + "start_code" + quoteColon;
+  private static final String serverNameInMeta = quote + "server_name_in_meta" + quoteColon;
+  private static final String listOfServers = quote + "list_of_servers" + quoteColon;
+  private static final String region1Info = quote + "region1_info" + quoteColon;
+  private static final String region2Info = quote + "region2_info" + quoteColon;
+  private static final String regionInfo = quote + "region_info" + quoteColon;
+  private static final String serverName = quote + "server_name" + quoteColon;
+  private static final String tableName = quote + "table_name" + quoteColon;
+
+  private static final String dataStartsWith = "{\"data\":[";
+  private static final String dataEndsWith = "]}";
+  private static final String hbckReportStartTime = quote + "hbck_report_start_time" + quoteColon;
+  private static final String hbckReportEndTime = quote + "hbck_report_end_time" + quoteColon;
+  private static final String hbckOrphanRegionOnFS =
+    quote + "hbck_orphan_regions_on_fs" + quoteColon;
+  private static final String hbckOrphanRegionOnRS =
+    quote + "hbck_orphan_regions_on_rs" + quoteColon;
+  private static final String hbckInconsistentRegion =
+    quote + "hbck_inconsistent_regions" + quoteColon;
+  private static final String hbckHoles = quote + "hbck_holes" + quoteColon;
+  private static final String hbckOverlaps = quote + "hbck_overlaps" + quoteColon;
+  private static final String hbckUnknownServers = quote + "hbck_unknown_servers" + quoteColon;
+  private static final String hbckEmptyRegionInfo = quote + "hbck_empty_region_info" + quoteColon;
+
+  @ClassRule
+  public static final HBaseClassTestRule CLASS_RULE =
+    HBaseClassTestRule.forClass(TestHbckMetricsResource.class);
+
+  private static final MiniClusterRule miniClusterRule = MiniClusterRule.newBuilder()
+    .setMiniClusterOption(
+      StartTestingClusterOption.builder().numZkServers(3).numMasters(3).numDataNodes(3).build())
```

Comment:
do you need 3 servers, or would the default 1 be good enough?
```suggestion
      StartTestingClusterOption.builder().build())
```

Reply:
I have used similar configs from other tests for number of master and regionserver. I will look into more tests to see if it's beneficial to reduce number HM/RS and create JIRA if it's helpful.